src/NHibernate/Hql/Ast/ANTLR/QueryTranslatorImpl.cs

```diff
@@ -430,6 +451,50 @@ private void ErrorIfDML()
 				throw new QueryExecutionRequestException("Not supported for DML operations", _queryIdentifier);
 			}
 		}
+
+		private bool ContainsRestrictionOnTable(string tableAlias)
```

Comment:
Can't this method be simplified to find child node with 
`child.FromElement == fromElement`

Reply:
Oups. Sorry - I made a quick debug and didn't actually check if it's really interface method. It seems casting to `FromReferenceNode' is required. See https://github.com/nhibernate/nhibernate-core/pull/463/commits/2f66408bafe95c9a45e7c9d6137912c5a713ed95 to give the idea. As I know nothing about HQL code -  I'm not sure it's the best way to traverse through children elements.